frontend/components/password-change/js/password-change.component.ts

```diff
@@ -105,36 +113,26 @@ class PasswordChangeController implements ng.IController {
 
 	}
 
+	public confirmPasswordInput() {
+		return this.registerMessage = this.PasswordService.checkDuplicates(this.newPassword, this.confirmPassword);
```

Comment:
This is super overkill. We don't need to go all the way to password service to do an equality check.

avoid class scope variable unless it's absolutely justified. 

All these can be done in the pug with an ngIf. i don't think this function needs to exist.

Reply:
I completley agree :) as i was refactoring i have already removed it from the service.